tools/topology/topology2/platform/intel/dmic-default.conf

```diff
@@ -43,5 +43,7 @@ Define {
 
 	# TDFB and DRC pipeline
 	DMIC0_ENHANCED_CAPTURE		"false"
-	DMIC0_PCM_CHANNELS		2
+
+	# Note: This will be redefined in dmic-generic.conf if not set from cmake
+	DMIC0_PCM_CHANNELS		0
```

Comment:
I guess a naive
```
    DMIC0_PCM_CHANNELS NUM_DMICS
```
wouldn't work?

Reply:
that was my initial proposal (https://github.com/thesofproject/sof/pull/9368#discussion_r1719489513) but it looks like the language does not  accept it.

Reply:
Yep, it's not possible (DMIC0_PCM_CHANNELS $NUM_DMICS).